neqo-transport/src/connection.rs

```diff
@@ -5173,4 +5183,141 @@ mod tests {
         client.process_input(Datagram::new(loopback(), loopback(), vec![77; 21]), now());
         assert!(matches!(client.state(), State::Draining { .. }));
     }
+
+    #[test]
+    fn server_receive_unknown_first_packet() {
+        let mut server = default_server();
+
+        let mut unknown_version_packet = vec![0x80, 0x1a, 0x1a, 0x1a, 0x1a];
+        unknown_version_packet.resize(1200, 0x0);
+
+        assert_eq!(
+            server.process(
+                Some(Datagram::new(
+                    loopback(),
+                    loopback(),
+                    unknown_version_packet,
+                )),
+                now(),
+            ),
+            Output::None
+        );
+
+        assert_eq!(1, server.stats().dropped_rx);
+    }
+
+    fn get_vn(client: &mut Connection, versions: &[u32]) -> Vec<u8> {
+        // Start the handshake.
+        let out = client
+            .process(None, now())
+            .dgram()
+            .expect("a datagram")
+            .to_vec();
+
+        let mut dec = Decoder::from(&out[5..]); // Skip past version.
+        let dcid = dec.decode_vec(1).expect("client DCID");
+        let scid = dec.decode_vec(1).expect("client SCID");
+
+        let mut encoder = Encoder::default();
+        encoder.encode_byte(PACKET_BIT_LONG);
+        encoder.encode(&[0; 4]); // Zero version == VN.
+        encoder.encode_vec(1, dcid);
+        encoder.encode_vec(1, scid);
+
+        for v in versions {
+            encoder.encode_uint(4, *v);
+        }
+        encoder.into()
+    }
+
+    #[test]
+    fn version_negotiation_current_version() {
+        let mut client = default_client();
+
+        let vn = get_vn(&mut client, &[0x1a1_a1a1a, QUIC_VERSION]);
+
+        assert_eq!(
+            client.process(Some(Datagram::new(loopback(), loopback(), vn,)), now(),),
```

Comment:
This is an odd time to be processing a VN packet.  Can you make this a little more conventional in the sense that it is received after sending the first Initial?

Reply:
The initial is sent inside get_vn(). I moved it there because we need the Intitial packet to retrieve cids.

Reply:
OK, that's misunderstanding on my part.  Renaming get_vn to handle_vn might work, but you would have to have it return the result of the client.process() call. The advantage of that is that you save another line of code from every test.